src/angular-breadcrumb.js

```diff
@@ -42,8 +42,13 @@ function $Breadcrumb() {
         // Get the parent state
         var $$parentState = function(state) {
             // Check if state has explicit parent OR we try guess parent from its name
-            var name = state.parent || (/^(.+)\.[^.]+$/.exec(state.name) || [])[1];
+            var parent = state.parent || (/^(.+)\.[^.]+$/.exec(state.name) || [])[1];
             // If we were able to figure out parent name then get this state
+            var name = parent;
+            if( typeof parent === "object" ) {
+                // if parent is a object reference, then extract the name
+                name = parent.name;
+            }
```

Comment:
How about a ternary operator like [this one](https://github.com/ncuillery/angular-breadcrumb/blob/master/src/angular-breadcrumb.js#L87) ?